openedx/core/djangoapps/course_roles/README.rst

```diff
@@ -0,0 +1,15 @@
+Course Roles
+#############################
+
+Status: Development
+
+Purpose
+*******
+
+The intention of this app is to to make course level roles for users within LMS/CMS more flexible. 
```

Comment:
Are there software patterns that this work enforces and anti-patterns that it prohibits. If so, calling this out is a stronger statement than just the pursuit of flexibility.  For example, a web page of benefits of role-based access control for Azure reads: "you can segregate duties within your team and grant only the amount of access to users that they need to perform their jobs."